feat: partial import export (#1939)
appsmithorg/appsmith-docs: website/docs/build-apps/how-to-guides/import-export-app-entities.md

---
description: A guide that outlines steps on how to export entities from a page to a JSON file and subsequently import it into another page.
---

> # Reuse App Entities

Author: Google Style guide violation. The allowed word is to use App instead of Application. Do we want to add an exception for this?

> # Reuse App Entities

Reviewer: Why is Application not allowed as a word? Yes, let's please add an exception

> # Reuse App Entities

Author: Google's wordlist. I've added an exception.

When working on complex Appsmith applications, you might find the need to reuse a particular set of widgets, queries, JSObjects, and more across pages. Instead of recreating these entities from scratch each time, Appsmith provides a way for you to export entities from one page and import them into another. This guide outlines steps on how to export entities from a page to a JSON file and subsequently import it into another page.

> ## Export entities

Reviewer: There is no need for subsections in this guide because it is a complete guide. It's not an option. As in they have to export followed by import. Also because users don't understand that we facilitate reuse by import and export, giving those as headings don't make sense.

Follow these steps to export your desired entities:

> * Open the Appsmith editor and access the app containing the page you wish to export.

Reviewer: i think this step is unecessary

```suggestion
```

Author: This is needed if they want to change the app. Will try to reword.

> * Click the three vertical dots next to the page name for more actions, and select the **Export** option from the dropdown menu. A dialog box will appear, allowing you to choose the entities you want to reuse.

Reviewer: ```suggestion
* Open the page from which you want to export the entities. Click the vertical dots icon next to the page name for the action menu.

* Select the **Export** option from the dropdown menu. A dialog box will appear, allowing you to choose the entities you want to reuse.
```

Author: modified a bit.

> <figure>

Reviewer: remove this, add an arcade at the top

Author: I don't think arcade is needed here, as we are just showing them clicks on entities that don't add any value to the user. A screenshot should be okay here.

  <img src="/img/reuse-entities-across-pages-export.png" style= {{width:"640px", height:"auto"}} alt="Export entities"/>
  <figcaption align = "center"><i>Select entities to export</i></figcaption>
</figure>   

> * Make your selections and click **Export selected entities** to download the JSON file of your entities. The JSON file follows the naming convention - `application name.json`.

Reviewer: ```suggestion
* Select the specific entities under each category and click **Export selected entities** to download the JSON file of your entities. The JSON file follows the naming convention - `application name.json`.
```

> ## Import entities

Reviewer: ```suggestion
```

> With the definitions of entities downloaded into a JSON file, you can import them to another page. Follow these steps to import the entities:

Reviewer: ```suggestion

```

> :::info

Reviewer: ```suggestion
:::caution
```

Author: I don't think this is caution.

The JSON file is a blueprint of your selected entities, and designed for sharing entities across pages. The JSON file can't be used to create a new app. 
:::

> * Open the editor and access the app or page where you wish to add your exported entities.

Reviewer: ```suggestion

```

> * Click the three vertical dots next to the page name for more actions, and select the **Import** option from the dropdown menu. 

Reviewer: ```suggestion
* Open the page where you want to import the entities. Click the vertical dots icon next to the page name for the action menu.
```

>  * A dialog box appears. You may choose to drag and drop or go to the location of your exported JSON file on your system and upload it. Upon completion of the import process, the entities from the JSON file are available on the selected page.

Reviewer: ```suggestion
 * Select the **Import** option from the dropdown menu. A dialog box appears. You may choose to drag and drop or go to the location of your exported JSON file on your system and upload it. Upon completion of the import process, the entities from the JSON file will be available on the current page.
```

> As a security measure, Appsmith does not export/import any configuration values used for connecting to the datasource. After importing, you will have to manually configure the datasource for the newly imported entities.

Reviewer: should be a caution callout

```suggestion
As a security measure, Appsmith does not export or import any configuration values used for connecting to a datasource. After importing, you will have to manually configure the datasource for the newly imported entities.
```

---
description: A guide that outlines steps on how to export entities from a page to a JSON file and subsequently import it into another page.
---

# Reuse Application Entities

This guide shows how to export certain entities from a page and import them into another page or application. Reusable entities are widgets, datasources, queries, JS Objects, and custom libraries. 

Follow the steps below to export desired application entities and import it into another page or an application.

1. Open the page from which you want to export the entities. Click the vertical dots icon next to the page name for the action menu.

<figure>
 <img src="/img/reuse-entities-across-pages-action-menu.png" style= {{width:"640px", height:"auto"}} alt="Export entities"/>
  <figcaption align = "center"><i>Action menu for import and export</i></figcaption>
</figure> 

2. Select the **Export** option from the action menu. A dialog box will appear, allowing you to choose the entities you want to reuse.
<figure>

>   <img src="/img/reuse-entities-across-pages-import-export.png" style= {{width:"640px", height:"auto"}} alt="Import or Export entities"/>

Reviewer: One image will suffice like this
<img width="354" alt="Screenshot 2023-12-06 at 14 08 13" src="https://github.com/appsmithorg/appsmith-docs/assets/32227135/c76e3eef-f96e-4921-82b8-39dbcf530002">
